### ISSUE TITLE
Add source/sink mismatch exceptions

### ISSUE THREAD

**Issue by EliLazarus** [julia-mpsge/MPSGE.jl#163]
If a user includes a variable which is incorporated as either a source (endowment/input) or sink (final demand/outputs) but missing the reciprocal use, we should generate a specific error message identifying the mismatch (at the build stage).
Related to #38

**Comment by mitchphillipson** [comment]
I don't think we can do this at build because the model builds incrementally. Perhaps on first solve we can run a check. 

There is something about market clearance here, even solving a counterfactual this would only balance if the commodity were zero. I'll do some testing soon.

**Comment by mitchphillipson** [comment]
Benchmarks are verified with 

```juila
solve!(M, cumulative_iteration_limit=0)
```
and checking the residual value. We could implement a `benchmark_validation` function, but it's going to be the same. Better would be to extend a `JuMP.is_solved_and_feasible` to an MPSGE model.
